web-common/src/features/sources/saveAndRefresh.ts

```diff
@@ -4,7 +4,10 @@ import { runtime } from "../../runtime-client/runtime-store";
 import { getFileAPIPathFromNameAndType } from "../entity-management/entity-mappers";
 import { EntityType } from "../entity-management/types";
 
-export async function saveAndRefresh(tableName: string, yaml: string) {
+export async function saveAndRefresh(
+  tableName: string,
+  yaml: string | undefined
```

Comment:
`yaml` should always be string, right? If it happens to be `undefined`, can we disable the Button instead?